### PR TITLE
Add testing for the lifetime of BLS tensors

### PR DESCRIPTION
**Pull request by Tabrizian**
https://github.com/triton-inference-server/python_backend/pull/131

### CODE REVIEW

File: qa/python_models/bls/model.py [hunk 1]
```diff
@@ -248,6 +248,58 @@ def test_zero_length_io(self):
...
+            infer_request = pb_utils.InferenceRequest(
+                model_name=model_name,
+                inputs=[input0_pb],
+                requested_output_names=['OUTPUT0'])
+            infer_response = infer_request.exec()
+            self.assertFalse(infer_response.has_error())
+
+            output0 = pb_utils.get_output_tensor_by_name(
+                infer_response, 'OUTPUT0')
+            self.assertTrue(np.all(output0.as_numpy() == input0))
```
Comment (dyastremsky) [inline qa/python_models/bls/model.py:272]: There's probably a better assertion for this. Maybe [numpy.testing.assert_equal](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_equal.html) for a clearer failure message?

File: qa/python_models/bls/model.py [hunk 2]
```diff
@@ -248,6 +248,58 @@ def test_zero_length_io(self):
...
+
+            output0 = pb_utils.get_output_tensor_by_name(
+                infer_response, 'OUTPUT0')
+            output0_pytorch = from_dlpack(output0.to_dlpack())
+
+            # Set inference response and output0_pytorch to None, to make sure
+            # that the DLPack is still valid.
+            output0 = None
+            infer_response = None
+            self.assertTrue(torch.all(output0_pytorch == input0))
```
Comment (dyastremsky) [inline qa/python_models/bls/model.py:301]: You may want to add an error message to provide the input/output values when this fails.

**Review by tanmayv25 — DISMISSED** [review]

**Comment by tanmayv25** [comment]
Looks good to me. Address the David's comment though.

File: qa/python_models/bls/model.py (hunk 1 above)
Comment (Tabrizian) [reply qa/python_models/bls/model.py]: updated

File: qa/python_models/bls/model.py (hunk 2 above)
Comment (Tabrizian) [reply qa/python_models/bls/model.py]: updated

**Review by dyastremsky — APPROVED** [review]

**Review by tanmayv25 — APPROVED** [review]

**Pull request merged**